src/tree/map.rs

```diff
@@ -636,17 +625,11 @@ macro_rules! bound_setup {
      // whether we are looking for the lower or upper bound.
      $is_lower_bound:expr) => {
         {
-            // FIXME: redundant, but a bug in method-level where clauses requires it
-            fn compare<C, Q: ?Sized, K>(cmp: &C, k: &Q, node_k: &K) -> Ordering
-                where C: Compare<Q, K> {
-                cmp.compare(k, node_k)
-            }
-
             let (mut iter, cmp) = $iter;
             loop {
                 if !iter.node.is_null() {
```

Comment:
I'm actually going to be replacing this pointer stuff with a safe reference-based `Iter` and a `transmute`-based `IterMut` in a future PR. This should also allow us to remove the macros.